### PR TITLE
Drop Python 3.3 and crash installation on unsupported versions of Python

### PR DESCRIPTION
**Pull request by dargueta**
Fixes issue https://github.com/CloverHealth/pytest-pgsql/issues/4 and drops support for Python 3.3.

The manual version check in `setup.py` is unfortunately necessary for versions of `pip` before 9.0, as described in [this warning](https://packaging.python.org/tutorials/distributing-packages/#python-requires) on python.org.

@wesleykendall

### CODE REVIEW

File: test_requirements.txt [hunk 1]
```diff
@@ -7,6 +7,11 @@
 pytest>=3.0.0; python_version>="3.4"
 pytest>=3.0,<3.3; python_version<"3.4"
 
+# isort is an indirect requirement of pylint. isort 4.3 dropped support
+# for Python 3.3, so we need to peg it to an earlier version to avoid
+# breakages.
+isort==4.2.15; python_version<"3.4"
```
Comment (wesleykendall) [inline test_requirements.txt]: We're doing a lot of things to support python 3.3, which is no longer even supported by pytest and a lot of other tools. I think we should drop 3.3 support altogether

File: ChangeLog [hunk 2]
```diff
@@ -1,6 +1,9 @@
 CHANGES
 =======
 
+* Add note in README about Python 2 compatibility
+* Make uninstallable on Python < 3.3
+
```
Comment (wesleykendall) [inline ChangeLog]: This will be overwritten on the deploy to your squashed commit message

Comment (dargueta) [reply ChangeLog]: I didn't do that. o__O

File: test_requirements.txt (hunk 1 above)
Comment (dargueta) [reply test_requirements.txt]: Okay. I'll drop 3.3 support in this PR.

**Comment by wesleykendall** [comment]
@dargueta this LGTM. If you are good to go, I'll deploy. Do you want a minor or patch version bump?

**Comment by dargueta** [comment]
Judging from the FAQ [here](https://semver.org/) it should probably be a minor version bump.

**Comment by dargueta** [comment]
@wesleykendall I think you can close issue #4 now BTW.

**Pull request merged**